cmp/path.go

```diff
@@ -94,10 +94,19 @@ func (pa *Path) pop() {
 // Last returns the last PathStep in the Path.
 // If the path is empty, this returns a non-nil PathStep that reports a nil Type.
 func (pa Path) Last() PathStep {
-	if len(pa) > 0 {
-		return pa[len(pa)-1]
+	return pa.Index(-1)
+}
+
+// Index returns the ith step in the Path and supports negative indexing.
```

Comment:
Pedantic: While anyone familiar with Python knows what negative indexing is, if you want to be precise you should define a negative index.

Reply:
Done.